common/djangoapps/edxmako/shortcuts.py

```diff
@@ -51,7 +51,10 @@ def marketing_link(name):
     elif not enable_mktg_site and name in link_map:
         # don't try to reverse disabled marketing links
         if link_map[name] is not None:
-            return reverse(link_map[name])
+            if 'http' in link_map[name]:
```

Comment:
To reduce the risk of false positives, we should check not only that the protocol is in the text, but that the text starts with the protocol.

And actually, I'm okay with restricting this to only support external links via SSL (`https` instead of `http`), so we could do something like:

```python
if link_map[name].startswith('https://'):
    return link_map[name]
```